src/views/RequiredSignoffs/ViewSignoff/index.jsx

```diff
@@ -29,106 +39,250 @@ const useStyles = makeStyles(theme => ({
   },
   fab: {
     ...theme.mixins.fab,
+    right: theme.spacing(12),
   },
   gridWithIcon: {
     marginTop: theme.spacing(3),
   },
+  addRoleButton: {
+    width: '100%',
+  },
+  addRoleGrid: {
+    marginTop: theme.spacing(5),
+  },
+  paper: {
+    position: 'absolute',
+    zIndex: 1,
+    marginTop: theme.spacing(1),
+    left: 0,
+    right: 0,
+  },
 }));
 
-function ViewSignoff(props) {
+function ViewSignoff({ isNewSignoff, ...props }) {
+  const emptyRole = ['', '', { isAdditionalRole: true, id: Math.random() }];
+  const { product, channel } = props.match.params;
   const classes = useStyles();
-  // eslint-disable-next-line no-unused-vars
-  const [products, setProducts] = useState();
-  // eslint-disable-next-line no-unused-vars
-  const [requiredSignoffs, setRequiredSignoffs] = useState(null);
-  const [error, setError] = useState();
-  const [type, setType] = useState('channel');
+  const [channelTextValue, setChannelTextValue] = useState(channel || '');
+  const [productTextValue, setProductTextValue] = useState(product || '');
+  const [type, setType] = useState(channel ? 'channel' : 'permission');
+  const [roles, setRoles] = useState([]);
+  const [additionalRoles, setAdditionalRoles] = useState(
+    isNewSignoff ? [emptyRole] : []
+  );
+  const [requiredSignoffs, getRS] = useAction(getRequiredSignoffs);
+  const [products, getProds] = useAction(getProducts);
+  const isLoading = requiredSignoffs.loading || products.loading;
   const handleTypeChange = ({ target: { value } }) => setType(value);
+  const handleChannelChange = ({ target: { value } }) =>
+    setChannelTextValue(value);
+  const handleProductChange = value => setProductTextValue(value);
+  const handleRoleValueChange = (role, index) => ({ floatValue: value }) => {
+    const setRole = (entry, i) =>
+      i === index ? [entry[0], value, entry[2]] : entry;
+
+    return role[2].isAdditionalRole
+      ? setAdditionalRoles(additionalRoles.map(setRole))
+      : setRoles(roles.map(setRole));
+  };
+
+  const handleRoleNameChange = (role, index) => ({ target: { value } }) => {
+    const result = additionalRoles.map((entry, i) =>
+      i === index ? [value, entry[1], entry[2]] : entry
+    );
+
+    setAdditionalRoles(result);
+  };
+
+  const handleRoleAdd = () => {
+    setAdditionalRoles(additionalRoles.concat([emptyRole]));
+  };
+
+  const handleRoleDelete = (role, index) => () => {
+    const excludeRole = (entry, i) => !(i === index);
+
+    return role[2].isAdditionalRole
+      ? setAdditionalRoles(additionalRoles.filter(excludeRole))
+      : setRoles(roles.filter(excludeRole));
+  };
+
+  // TODO: Add save logic
+  const handleSignoffSave = () => {};
+  // TODO: Add delete logic
+  const handleSignoffDelete = () => {};
```

Comment:
@mozbhearsum I was thinking the mutations (hooking up the actions) is probably something you can do. This would allow me to work on the general layout of the other views. Let me know what you think. I plan to tackle the rules page next. Just need to find the time :)

Reply:
That's a great idea. Would you mind doing just one of them first, though? I still don't have a great feel for the structure of React apps, so that would help me bootstrap.

Reply:
Sure thing. I can try to write something up in a separate PR.